service/history/historyEngine.go

```diff
@@ -2194,6 +2194,16 @@ func (e *historyEngineImpl) ReplicateEvents(replicateRequest *h.ReplicateEventsR
 	return e.replicator.ApplyEvents(replicateRequest)
 }
 
+func (e *historyEngineImpl) SyncShardStatus(ctx context.Context, request *h.SyncShardStatusRequest) error {
+	clusterName := request.GetSourceCluster()
+	now := time.Unix(0, request.GetTimestamp())
+
+	e.shard.SetCurrentTime(clusterName, now)
+	e.txProcessor.NotifyNewTask(clusterName, now, []persistence.Task{})
```

Comment:
Do we really need this notification here?  I think we already have protection to trigger scan at regular bases.  Firing a notification to force a scan sounds redundant.

Reply:
talked offline, this should be fine